### PR TITLE
Support filtering specified minion jobs

### PR DESCRIPTION
**Pull request by Amrysliu**
Related: https://progress.opensuse.org/issues/70768

### CODE REVIEW

File: lib/OpenQA/Setup.pm [hunk 1]
```diff
@@ -35,26 +35,27 @@ sub read_config {
...
+            plugins                       => undef,
+            hide_asset_types              => 'repo',
+            recognized_referers           => '',
+            changelog_file                => '/usr/share/openqa/public/Changelog',
+            job_investigate_ignore        => '"(JOBTOKEN|NAME)"',
+            job_investigate_git_timeout   => 20,
+            worker_timeout                => DEFAULT_WORKER_TIMEOUT,
+            search_results_limit          => 50000,
+            auto_clone_regex              => '^cache failure: ',
+            influxdb_minion_job_blocklist => undef,
```
Comment (Amrysliu) [inline lib/OpenQA/Setup.pm:58]: I only add `influxdb_minion_job_blocklist => undef`

File: t/api/13-influxdb.t [hunk 2]
```diff
@@ -67,6 +67,28 @@ $t->get_ok('/admin/influxdb/minion')->status_is(200)->content_is(
...
+
+    $t->get_ok('/admin/influxdb/minion')->status_is(200)->content_is(
+        "openqa_minion_jobs,url=http://example.com active=0i,delayed=1i,failed=3i,inactive=1i
+openqa_minion_workers,url=http://example.com active=0i,inactive=1i
+"
+    );
+
+    $t->app->config->{global}->{influxdb_minion_job_blocklist} = 'obs_rsync_run obs_rsync_update_builds_text';
+    $t->get_ok('/admin/influxdb/minion')->status_is(200)->content_is(
+        "openqa_minion_jobs,url=http://example.com active=0i,delayed=1i,failed=1i,inactive=1i
```
Comment (kalikiana) [inline t/api/13-influxdb.t:86]: So this contains `failed=1i` in place of `failed=3i`. It's very difficult to even see that they are different. Could you please add a comment pointing out the difference and why this is correct?

**Review by kalikiana — CHANGES_REQUESTED** [review]

File: t/api/13-influxdb.t (hunk 2 above)
Comment (Amrysliu) [reply t/api/13-influxdb.t:88]: added

**Review by kalikiana — APPROVED** [review]

File: t/api/13-influxdb.t [hunk 3]
```diff
@@ -67,6 +67,30 @@ $t->get_ok('/admin/influxdb/minion')->status_is(200)->content_is(
...
+    }
+    while (my $tmp_job = $worker->dequeue(0)) { $tmp_job->fail('this is a test'); }
+
+    $t->get_ok('/admin/influxdb/minion')->status_is(200)->content_is(
+        "openqa_minion_jobs,url=http://example.com active=0i,delayed=1i,failed=3i,inactive=1i
+openqa_minion_workers,url=http://example.com active=0i,inactive=1i
+"
+    );
+
+    # Define the filter list, so the failed jobs whose task belong to the list
```
Comment (kalikiana) [inline t/api/13-influxdb.t:84]:
```suggestion
    # Define the filter list, so failed jobs for blocklisted tasks will not be
    # counted towards the total of failed jobs, i.e. failed=1i instead of failed=3i
```

**Review by okurz — COMMENTED** [review]
+1

what cdywan said

**Review by kalikiana — APPROVED** [review]

File: lib/OpenQA/WebAPI/Controller/Admin/Influxdb.pm [hunk 4]
```diff
@@ -107,11 +107,15 @@ sub jobs {
 sub minion {
     my $self = shift;
 
-    my $stats = $self->app->minion->stats;
-    my $jobs  = {
+    my $stats           = $self->app->minion->stats;
+    my $block_list      = $self->app->config->{global}->{influxdb_minion_job_blocklist} || '';
+    my @filter_task     = split(/\s+/, $block_list);
```
Comment (Martchus) [inline lib/OpenQA/WebAPI/Controller/Admin/Influxdb.pm:112]: I would do the splitting when reading the settings and not every time the route is accessed.

File: lib/OpenQA/Setup.pm (hunk 1 above)
Comment (Martchus) [reply lib/OpenQA/Setup.pm:58]: Maybe influxdb deserves is own section although we only have one setting so far. It would also be nice to extend our sample/default config file. The name of the key should maybe indicate that this is about failed jobs.

File: lib/OpenQA/WebAPI/Controller/Admin/Influxdb.pm (hunk 4 above)
Comment (Amrysliu) [reply lib/OpenQA/WebAPI/Controller/Admin/Influxdb.pm]: Put influxdb_minion to its own section, also do the splitting when reading settings.

File: t/config.t [hunk 5]
```diff
@@ -193,8 +194,6 @@ subtest 'trim whitespace characters from both ends of openqa.ini value' => sub {
     ';
     $t_dir->child('openqa.ini')->spurt($data);
     OpenQA::Setup::read_config($app);
-    print $app->config->{global}->{appname};
-    print $app->config->{global}->{hide_asset_types};
```
Comment (okurz) [inline t/config.t:197]: unrelated changes (including the whitespace changes above). Better create a separate commit. Can still be part of this PR but separate commit.

File: etc/openqa/openqa.ini [hunk 6]
```diff
@@ -159,3 +159,7 @@ blocklist = job_grab job_done
 # executable name in `$PATH` is called with the job ID as first and only
 # parameter corresponding to the `$result`, for example
 #job_done_hook_failed = my-openqa-hook-failed
+
+## Specify minion failed job task name. The failed jobs for blocklisted tasks will not be counted towards the total of failed jobs.
+[influxdb_minion_fail_job]
+#blocklist = influxdb-minion-fail-job-task-name
```
Comment (Martchus) [inline etc/openqa/openqa.ini:165]: I would have structured this a little bit different. My intention with the first review comment was to have an own section for everything InfluxDB related. I would also use past here (fail**ed**). I also shortened the comment a little bit.

```suggestion
# Configuration for InfluxDB routes
[influxdb]
# Specify Minion task names which should never be counted towards the total of number failed Minion jobs.
#ignored_failed_minion_jobs = influxdb-minion-fail-job-task-name
```

**Review by Martchus — CHANGES_REQUESTED** [review]

**Review by okurz — APPROVED** [review]

**Comment by kalikiana** [comment]
@Amrysliu Glancing at the failures

```
    #   Failed test 'Allocated maximum number of jobs that could have been allocated'
    #   at t/05-scheduler-full.t line 241.
    #          got: '9'
    #     expected: '10'
    # Looks like you failed 1 test of 1.
```

I think this might be flaky, but

````
t/api/04-jobs.t                  (Wstat: 14 Tests: 62 Failed: 0)
  Non-zero wait status: 14
```

looks fishy 🤔

File: t/config.t [hunk 7]
```diff
@@ -224,4 +223,20 @@ subtest 'Validation of worker timeout' => sub {
     };
 };
 
+subtest 'parse ignored_failed_minion_jobs' => sub {
```
Comment (Martchus) [inline t/config.t:226]: It would be less effort to extend the existing subtest `Test configuration override from file` instead of creating a new, similarly structured subtest for just one field.

**Review by Martchus — COMMENTED** [review]
The change itself looks good. I would have written the test differently (see comment) but that's not a blocker for me.

File: t/config.t (hunk 7 above)
Comment (kalikiana) [reply t/config.t:226]: Note that you're still "requesting changes" from a previous comment.

Comment (Amrysliu) [reply t/config.t]: fixed.

**Review by Martchus — APPROVED** [review]
